.github/workflows/main.yml

```diff
@@ -54,3 +55,27 @@ jobs:
         with:
           name: python-artifacts
           path: .
+  build-image:
+    name: Build and Push Image
+    needs: ['python-ci']
+    if: github.ref_name == 'main'
```

Comment:
For pull requests from a fork, the branch from which the PR is created has value of` github.ref_name` not equal `main ` branch, so this condition is not met and actions is skipped
```suggestion
    if: github.base_ref == 'main'
```